client/src/modules/Course/contexts/SessionContext.tsx

```diff
@@ -62,6 +63,20 @@ export function SessionProvider(props: Props) {
     );
   }
 
+  if (session && props.hirerOnly && !session.isHirer && !session.isAdmin) {
+    return (
+      <Result
+        status="warning"
+        title="You don't have required role to access this page"
+        extra={
+          <Button type="primary" key="console" onClick={() => window.history.back()}>
+            Go Back
+          </Button>
+        }
+      />
```

Comment:
this component was duplicated 2 times inside of the context, let's extract it and reuse instead of duplicating it 3rd time :)